consensus/vbft/service.go

```diff
@@ -1043,6 +1055,32 @@ func (self *Server) onConsensusMsg(peerIdx uint32, msg ConsensusMsg, msgHash com
 			fromPeer: peerIdx,
 			msg:      msg,
 		}
+	case BlockSubmitMessage:
+		pMsg, ok := msg.(*blockSubmitMsg)
+		if !ok {
+			log.Error("invalid msg with submit msg type")
+			return
+		}
+		msgBlkNum := pMsg.GetBlockNum()
+		if msgBlkNum > self.GetCurrentBlockNo() {
+			log.Errorf("failed to get submit msg (%d) to currentblock:%d", msgBlkNum, self.GetCurrentBlockNo())
+			return
+		}
+		if err := self.msgPool.AddMsg(msg, msgHash); err != nil {
+			if err != errDropFarFutureMsg {
+				log.Errorf("failed to add submit msg (%d) to pool: %s", msgBlkNum, err)
+			}
+			return
+		}
+		submitNum := self.msgPool.GetBlockSubmitMsgNums(msgBlkNum)
+		m := len(self.config.Peers) - (len(self.config.Peers)*3)/7
+		if submitNum < uint32(m) {
+			return
+		}
+		log.Infof("submit msg  submitNum:%d,m:%d", submitNum, m)
+		if err := self.chainStore.SubmitBlock(msgBlkNum); err != nil {
+			log.Errorf("SubmitBlock err:%s", err)
+		}
```

Comment:
where is the state root comparison between remote's and local's?

Reply:
already added  in next commit